donut/modules/directory_search/templates/search_results.html

```diff
@@ -39,6 +37,25 @@
 					{% endfor %}
 				</tbody>
 			</table>
-		{% endif %}
+			<ul id='demo' class="pagination">
+	<li class="page-item {% if page == 1 %} disabled {% endif %}">
+		<a class="page-link" {% if page != 1 %} href="{{ url_for('directory_search.search', page=page-1, total=total, num_pgs=num_pgs, name=name, house_id=house_id, option_id=option_id, building_id=building_id, grad_year=grad_year, username=username, email=email) }}" {% endif %} aria-label="Previous">
+        <span aria-hidden="true">&laquo;</span>
+        <span class="sr-only">Previous</span>
+      </a>
+    </li>
+    {% for i in range(1, num_pgs+1) %}
+    <li class="page-item {% if i == page %} active {% endif %}">
+	    <a class="page-link" href="{{ url_for('directory_search.search', page=i, total=total, num_pgs=num_pgs, name=name, house_id=house_id, option_id=option_id, building_id=building_id, grad_year=grad_year, username=username, email=email) }}">{{ i }}</a>
```

Comment:
Don't think `num_pgs` is needed

Reply:
I think it's needed so that it doesn't have to recalculate the number of pages when it's just switching pages

Reply:
Hmm, I see. I think you should be able to calculate `num_pgs` from `total`, though. That would decrease the number of parameters that need to get passed between the client and the server.